File: 2000-reverse-prefix-of-word/2000-reverse-prefix-of-word.java
class Solution {
    public String reversePrefix(String word, char ch) {
        
        for(int i =0;i<word.length();i++){
            
            if(word.charAt(i) == ch){
                int s = i+1;
                StringBuilder sb = new StringBuilder();
                while (i >= 0){
                    sb.append(word.charAt(i));
                    i--;
                }
                while(s < word.length()){
                    sb.append(word.charAt(s));
                    s++;
                }
                return sb.toString();
            }
        }
        return word;
    }
}

File: 2000-reverse-prefix-of-word/2000-reverse-prefix-of-word.py
class Solution:
    
    
    def reversePrefix(self, word: str, ch: str) -> str:
        
        l = 0
        for i in range(len(word)):
            
            if word[i] == ch:
                while(l < i):
                    word = list(word)
                    temp = word[l]
                    word[l] = word[i]
                    word[i]  = temp
                    l+=1
                    i-=1
                return "".join(word)
        
        return word


    
    
        